app/src/behaviors/behavior_hold_tap.c

```diff
@@ -200,34 +215,23 @@ static struct active_hold_tap *store_hold_tap(uint32_t position, uint32_t param_
 
 static void clear_hold_tap(struct active_hold_tap *hold_tap) {
     hold_tap->position = ZMK_BHV_HOLD_TAP_POSITION_NOT_USED;
-    hold_tap->is_decided = false;
-    hold_tap->is_hold = false;
+    hold_tap->status = STATUS_UNDECIDED;
     hold_tap->work_is_cancelled = false;
 }
 
-enum decision_moment {
-    HT_KEY_UP = 0,
-    HT_OTHER_KEY_DOWN = 1,
-    HT_OTHER_KEY_UP = 2,
-    HT_TIMER_EVENT = 3,
-    HT_QUICK_TAP = 4,
-};
-
 static void decide_balanced(struct active_hold_tap *hold_tap, enum decision_moment event) {
     switch (event) {
     case HT_KEY_UP:
-        hold_tap->is_hold = 0;
-        hold_tap->is_decided = true;
-        break;
+        hold_tap->status = STATUS_TAP;
+        return;
     case HT_OTHER_KEY_UP:
+        hold_tap->status = STATUS_HOLD_INTERRUPT;
+        return;
     case HT_TIMER_EVENT:
-        hold_tap->is_hold = 1;
-        hold_tap->is_decided = true;
-        break;
+        hold_tap->status = STATUS_HOLD_TIMER;
+        return;
     case HT_QUICK_TAP:
-        hold_tap->is_hold = 0;
-        hold_tap->is_decided = true;
-        break;
+        hold_tap->status = STATUS_TAP;
```

Comment:
Minor nitpick, can you add a `return;` to this case, in case `default` is changed later.